dockerpty/pty.py

```diff
@@ -249,3 +249,103 @@ def _hijack_tty(self, pumps):
                 except SSLError as e:
                     if 'The operation did not complete' not in e.strerror:
                         raise e
+
+
+class ExecPTY(PseudoTerminal):
+    def __init__(self, client, container, interactive=True, stdout=None, stderr=None, stdin=None):
+        """
+        Initialize the PTY using the docker.Client instance and container dict.
+        """
+        super(ExecPTY, self).__init__(client, container, interactive=interactive,
+                                      stdout=stdout, stderr=stderr, stdin=stdin)
+
+        self.exec_id = None
```

Comment:
Yes, it's possible to merge the class into `PseudoTerminal`. The reason I subclassed is that `PseudoTerminal` class is tied to running a container and attaching to it. `exec` is different operation with different workflow and API calls. If we merge those two use cases together, it will be pretty messy.